Makefile.am

```diff
@@ -361,18 +361,18 @@ functional-tests: $(FUNCTIONAL_TESTS_DEPS)
 			done; \
 		fi
 		@echo -e \
-		"export PROXY_SOCKET_PATH=\"$(abs_top_srcdir)/tests/functional/proxy.sock\" \n" \
-		"export SHIM_PATH=\"$(abs_top_srcdir)/cc-shim\" \n" \
+		"export PROXY_SOCKET_PATH=\"$(abs_builddir)/tests/functional/proxy.sock\" \n" \
```

Comment:
Like #599, I really think it would be better to put all this shell logic in a separate script (see my general comments on #599).